### PR TITLE
bernstein basis .py and test

### PR DESCRIPTION
**Pull request by clara-armstrong**
## Description
Bernstein basis .py and test. Hopefully, this version works.
## Linked Issue
Fixes # (5)

## Implementation Checklist
- [X] My class inherits from `PolynomialBasis` (if applicable).
- [X] I have implemented all abstract methods (`evaluate_basis`, `fit`, etc.).
- [X] I have used `numpy` vectorization to avoid inefficient Python loops.
- [X] Documentation strings (docstrings) are provided for all public methods.

### 1. Mathematical Integrity
- [ ] **Convergence:** Does the implementation maintain expected accuracy? (Checked for Runge's phenomenon or stability).
- [ ] **Edge Cases:** Does it handle single points or non-equally spaced nodes correctly?
- [ ] **Domain Mapping:** Does the class correctly map the input domain to the basis domain (e.g., [-1, 1] for Chebyshev)?

### 2. API & Style
- [ ] **Method Signatures:** Methods match the Abstract Base Class exactly.
- [ ] **Naming:** Follows `CamelCase` for classes and `snake_case` for functions/variables.
- [ ] **Clean Code:** No "magic numbers"; constants are named and explained.

### 3. Testing Quality
- [ ] **Coverage:** Unit tests cover typical usage and edge cases.
- [ ] **Precision:** Tests use `np.testing.assert_allclose` instead of `==`.
- [ ] **Validation:** There is a test proving the basis can exactly reconstruct a polynomial of its own degree.

## Screenshots / Plots

### CODE REVIEW

File: src/numanalysislib/basis/bernstein.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+            Returns:
+                Value of the Bernstein basis polynomial at x
+        """
+        if index < 0 or index > self.degree:
+            raise ValueError(f"Basis index {index} out of range for degree {self.degree}")
+        
+        # Ensure x is an array to handle scalar inputs correctly
+        x = np.asarray(x)
+        n = self.degree
```
Comment (mtezzele) [inline src/numanalysislib/basis/bernstein.py]: avoid using local variable the correspond to class member. Use directly `self.degree`

**Comment by mtezzele** [comment]
I think you are not using the latest version locally because your constructors do not use the new arguments. These tests should fail also on your local machine.

File: src/numanalysislib/basis/bernstein.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    def fit(self, x_nodes: np.ndarray, y_nodes: np.ndarray) -> np.ndarray:
+        """
+
+        Solving the linear system Ac = y to force interpolation
+
+    
+        Computes coefficients by solving A c = y,
+        where A_ij = B_{j,n}(x_i).
+
+        """
```
Comment (quialst) [inline src/numanalysislib/basis/bernstein.py:63]: Maybe add arguments and returns in your documentation.

File: src/numanalysislib/basis/bernstein.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            raise ValueError(f"Expected {self.n_dofs} nodes for degree {self.degree}, got {len(x_nodes)}")
+
+        n = self.degree
+
+        # Build Bernstein matrix A
+        A = np.zeros((self.n_dofs, self.n_dofs))
+        for i in range(self.n_dofs):
+            for j in range(self.n_dofs):
+                A[i, j] = comb(n, j) * (x_nodes[i] ** j) * ((1 - x_nodes[i]) ** (n - j))
+
```
Comment (quialst) [inline src/numanalysislib/basis/bernstein.py:76]: consider implementing Bernstein matrix logic using the evaluate basis function to improve code readability.

File: tests/test_bernstein_basis.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
+import numpy as np
+import pytest
+from numanalysislib.basis.bernstein import BernsteinBasis
+
+class TestBernsteinBasis:
+
```
Comment (quialst) [inline tests/test_bernstein_basis.py:7]: consider verifying that the Bernstein basis fitting is not vulnerable to high degrees of oscillation when fitting on nodes which are concentrated on the endpoints.

File: src/numanalysislib/basis/bernstein.py (hunk 2 above)
Comment (clara-armstrong) [reply src/numanalysislib/basis/bernstein.py:63]: Added!

**Comment by mtezzele** [comment]
I'm not sure how you rebased. I see the files from the integrator PR also here, so something went wrong...

File: src/numanalysislib/basis/bernstein.py (hunk 3 above)
Comment (mtezzele) [reply src/numanalysislib/basis/bernstein.py:76]: Agree, avoid code duplication

File: src/numanalysislib/basis/bernstein.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+Here we implement the Bernstein polynomial on the reference interval [0, 1]
+"""
+import warnings
+import numpy as np
+from math import comb
+from numanalysislib.basis._abstract import PolynomialBasis
+
+class BernsteinBasis(PolynomialBasis):
+    """  
+
```
Comment (mtezzele) [inline src/numanalysislib/basis/bernstein.py]: Remove unnecessary empty lines

File: src/numanalysislib/basis/bernstein.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    Implements the Bernstein Basis: {B_{0,n}, ..., B_{i,n}} on [0,1].
+
```
Comment (mtezzele) [inline src/numanalysislib/basis/bernstein.py]: Remove unnecessary empty lines
